db/views/materialized_patient_summaries_v03.sql

```diff
@@ -0,0 +1,416 @@
+    select distinct on (patient_id) *
+    from patient_business_identifiers
+    where identifier_type = 'simple_bp_passport' and deleted_at is null
+    order by patient_id, device_created_at desc
+), latest_phone_number as (
+    select distinct on (patient_id) *
+    from patient_phone_numbers
+    where deleted_at is null
+    order by patient_id, device_created_at desc
+), latest_medical_history as (
+    SELECT DISTINCT ON (patient_id) * FROM medical_histories WHERE deleted_at IS NULL
+),ranked_prescription_drugs as (
+    select
+        bp.id as bp_id,
+        prescription_drugs.*,
+        rank() over (partition by bp.id order by prescription_drugs.is_protocol_drug desc, prescription_drugs.name, prescription_drugs.device_created_at desc) rank
+    from blood_pressures bp
+    left outer join prescription_drugs
+        on prescription_drugs.patient_id = bp.patient_id
+        and (prescription_drugs.device_created_at < date(bp.recorded_at) + '1 day'::interval)
+        and (prescription_drugs.is_deleted is false or (prescription_drugs.is_deleted is true and prescription_drugs.updated_at >= date(bp.recorded_at) + '1 day'::interval))
+    where bp.deleted_at is null and prescription_drugs.deleted_at is null
+), other_medications as (
+    select
+        bp_id,
+        string_agg (name || '-' || dosage, ', ') filter (where rank > 5) as other_prescription_drugs,
+        string_agg (name || '-' || dosage, ', ' order by rank) as all_prescription_drugs
+    from ranked_prescription_drugs
+    group by bp_id
+), ranked_blood_pressures as (
+    select
+        bp.id,
+        bp.patient_id,
+        bp.recorded_at as recorded_at,
+        bp.systolic,
+        bp.diastolic,
+        f.name as facility_name,
+        f.facility_type,
+        f.district,
+        f.state,
+        follow_up_facility.name as follow_up_facility_name,
+        a.scheduled_date as follow_up_date,
+        greatest(0, date_part('day', a.scheduled_date - date_trunc('day', a.device_created_at)))::int as follow_up_days,
+
+        pd_1.name as prescription_drug_1_name,
+        pd_1.dosage as prescription_drug_1_dosage,
+
+        pd_2.name as prescription_drug_2_name,
+        pd_2.dosage as prescription_drug_2_dosage,
+
+        pd_3.name as prescription_drug_3_name,
+        pd_3.dosage as prescription_drug_3_dosage,
+
+        pd_4.name as prescription_drug_4_name,
+        pd_4.dosage as prescription_drug_4_dosage,
+
+        pd_5.name as prescription_drug_5_name,
+        pd_5.dosage as prescription_drug_5_dosage,
+
+        other_medications.other_prescription_drugs,
+        other_medications.all_prescription_drugs,
+
+        rank() over (partition by bp.patient_id order by bp.recorded_at desc) rank
+    from blood_pressures bp
+    left outer join facilities f on bp.facility_id = f.id
+    left outer join appointments a on a.patient_id = bp.patient_id and date_trunc('day', a.device_created_at) = date_trunc('day', bp.recorded_at)
```

Comment:
What if there are more than 1 appointments created on the same day of the BP. (this subquery can have multiple rows for the same BP)